app/controllers/application_controller.rb

```diff
@@ -12,14 +14,25 @@ def after_sign_out_path_for(resource)
     sign_in_path
   end
 
-  def check_subdomain?
+  def validate_subdomain
     subdomain = request.subdomain
-    if subdomain == 'www'
-      redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
-    elsif subdomain.blank? || Company.find_by(subdomain: subdomain).nil?
-      flash[:alert] = 'Get Registered First'
-      redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
+    if subdomain.present?
+      if subdomain == 'www'
+        redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
+      elsif Company.find_by(subdomain: subdomain).nil?
+        flash[:alert] = 'Get Registered First'
+        redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
+      end
+    end
+  end
+
+  def current_company
+    company = Company.find_by(subdomain: request.subdomain)
```

Comment:
make `company` it an instance variable so that we can use it in different parts of application